docs/sphinx/reference-sources.rst

```diff
@@ -945,7 +946,8 @@ General Source Functions
 
 .. function:: signal_handler_t *obs_source_get_signal_handler(const obs_source_t *source)
 
-   :return: The source's signal handler
+   :return: The source's signal handler. Should not be manually freed,
+            as its lifecycle is managed by the program.
```

Comment:
"the program" should probably be "the library" or "libobs" here for extra clarity.

Reply:
Please see new commit for the changes. If you approve, I'll squash it. I added the same clarification to similar signal handler and proc handlers.

Reply:
Looks good to me!